lib/src/screens/rescan/rescan_page.dart

```diff
@@ -9,24 +9,51 @@ import 'package:cake_wallet/src/screens/base_page.dart';
 import 'package:cake_wallet/src/widgets/blockchain_height_widget.dart';
 import 'package:cake_wallet/src/widgets/primary_button.dart';
 import 'package:cake_wallet/generated/i18n.dart';
+import 'package:cw_core/wallet_type.dart';
 
 class RescanPage extends BasePage {
-  RescanPage(this._rescanViewModel)
+  RescanPage(this._rescanViewModel, this.type)
       : _blockchainHeightWidgetKey = GlobalKey<BlockchainHeightState>();
 
   @override
   String get title =>
       _rescanViewModel.isSilentPaymentsScan ? S.current.silent_payments_scanning : S.current.rescan;
   final GlobalKey<BlockchainHeightState> _blockchainHeightWidgetKey;
   final RescanViewModel _rescanViewModel;
+  final WalletType type;
 
   @override
   Widget body(BuildContext context) {
+    if (type == WalletType.decred) {
+      return Center(
+          child: Padding(
+        padding: EdgeInsets.only(left: 24, right: 24, bottom: 24),
+        child: Column(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Spacer(),
+              Observer(
+                  builder: (_) => LoadingPrimaryButton(
+                        isLoading: _rescanViewModel.state ==
+                            RescanWalletState.rescaning,
+                        text: S.of(context).rescan,
+                        onPressed: () async {
+                          await _rescanViewModel.rescanCurrentWallet(
+                              restoreHeight: 0);
```

Comment:
It will rescan from the wallet birthday which is encoded in the 15 word seed used to create the wallet. This is good as long as no private data is imported into the wallet, which can't currently happen.